GrandCentralBoard/Widgets/Watch/Server/GoogleCalendarDataProvider.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  GoogleCalendarDataProvider.swift
+//  GrandCentralBoard
+//
+//  Created by Michał Laskowski on 07.04.2016.
+//  Copyright © 2016 Oktawian Chojnacki. All rights reserved.
+//
+
+import Foundation
+import Result
+
+protocol CalendarDataProvider {
+    func fetchEventsForCalendar(calendarID: String, completion: (Result<[EventModel], APIDataError>) -> Void)
+    func fetchCalendar(calendarID: String, completion: (Result<CalendarModel, APIDataError>) -> Void)
+}
+
+final class GoogleCalendarDataProvider : CalendarDataProvider {
+
+    let dataProvider: APIDataProvider
+
+    init(dataProvider: APIDataProvider) {
+        self.dataProvider = dataProvider
+    }
+
+    private static let dateFormatter: NSDateFormatter =  {
+        let formatter = NSDateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ";
+        return formatter
+    }()
+
+    func fetchEventsForCalendar(calendarID: String, completion: (Result<[EventModel], APIDataError>) -> Void) {
+        guard let escapedCalendarID = calendarID.URLEscape() else {
+            assertionFailure("Failure escaping calendar id: \(calendarID)")
+            return
+        }
+        let url = NSURL(string: "https://www.googleapis.com/calendar/v3/calendars/\(escapedCalendarID)/events")!
+        let timeMin = self.dynamicType.dateFormatter.stringFromDate(NSDate())
+        let parameters: [String : AnyObject] = ["maxResults" : 10, "orderyBy" : "startTime",
+                                                "singleEvents" : "true", "maxAttendees" : 1,
```

Comment:
We do not display attendees, and the minival value here is 1. So to minimize the response size I used the minimal allowed value. I will add a comment about it, ok?